experimental/ragged_inference/garbage_pad_ragged_acts.py

```diff
@@ -0,0 +1,191 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+from typing import List
+
+import numpy as np
+import torch
+import triton
+import triton.language as tl
+
+
+@triton.jit
+def garbage_pad_ragged_acts_kernel(
+    ragged_acts_ptr,
+    ragged_acts_offset_per_seq_ptr,
+    n_ctx_per_seq_ptr,
+    padded_acts_ptr,
+    **meta,  # Optional meta-parameters for the kernel
+):
+    BLOCK_SIZE = meta["d_model"]  # How many inputs each program should process
+    # There are multiple 'program's processing different data. We identify which program
+    # we are here
+
+    seq_idx = tl.program_id(axis=0)
+    ctx_idx = tl.program_id(axis=1)
+
+    # This program will process inputs that are offset from the initial data.
+    # for instance, if you had a vector of length 256 and block_size of 64, the programs
+    # would each access the elements [0:64, 64:128, 128:192, 192:256].
+
+    ragged_acts_offset_ptr = ragged_acts_offset_per_seq_ptr + seq_idx
+    ragged_acts_offset = tl.load(ragged_acts_offset_ptr)
+
+    # Create a mask to guard memory operations against out-of-bounds accesses
```

Comment:
Is this unavoidable? How much more memory does this consume (if any)?

Reply:
it's part of the Triton API (masked load), it should not take any registers really. This is all in SoC space, so nothing visible in the RAM, worst case it's a throwaway mask on the actual gpu chip